File: clojure/test/kozko2001/day02_test.clj
(ns kozko2001.day02-test
  (:require [clojure.test :refer :all]
            [kozko2001.day02 :refer :all]))

(def test-data-part1 (slurp "resources/day02-test-part1.txt"))
(def data-part1 (slurp "resources/day02-part1.txt"))

(deftest part1-puzzle
  (is (= 8 (part1 test-data-part1))))

(deftest part1-real
  (is (= 2776 (part1 data-part1))))

(deftest part2-puzzle
  (is (= 2286 (part2 test-data-part1))))

(deftest part2-real
  (is (= 68638 (part2 data-part1))))
(clojure.test/run-tests)

File: elixir/test/day02_test.exs
defmodule Day02Test do
  use ExUnit.Case

  test "part1 with testing data" do
    {:ok, content} = File.read("test/day02_testing.txt")
    assert Day02.part1(content) == 8
  end

  test "part1 with real data" do
    {:ok, content} = File.read("test/day02_real.txt")
    assert Day02.part1(content) == 2776
  end

  test "part2 with testing data" do
    {:ok, content} = File.read("test/day02_testing.txt")
    assert Day02.part2(content) == 2286
  end

  test "part2 with real data" do
    {:ok, content} = File.read("test/day02_real.txt")
    assert Day02.part2(content) == 68638
  end
end
